csrc/ir_nodes.cpp

```diff
@@ -1351,6 +1351,17 @@ MmaOp::MmaOp(
           in_b->getValType().value() == ValType::TensorIndex,
       in_b->getValType().value());
 
+  const auto isBroadcastIn = [](const Val* val) {
+    if (val->getValType().value() == ValType::TensorView) {
+      const auto* tv = val->as<TensorView>();
+      return tv->hasBroadcast();
+    }
+    return true;
+  };
+
+  TORCH_INTERNAL_ASSERT(isBroadcastIn(in_a));
+  TORCH_INTERNAL_ASSERT(isBroadcastIn(in_b));
```

Comment:
Note for the future:

I think we should not only check the input has a broadcast, but also to add many more consistency check. For this PR, I think it is fine, but let's create a separate followup PR for it:

I think we should check:
- Inputs has two concrete IDs and one broadcast ID
- The broadcast's axis in different inputs are different
- Output has two concrete IDs and one reduction ID
- The axis of the output reduction ID must correspond to a concrete ID in both inputs

And with this check here, we can remove corresponding checks in the scheduler.